### PR TITLE
Added new common issue for missing SAML attributes

### PR DESCRIPTION
**Pull request by bontreger**
Link references Pull Request #22 
Tag: @itewk

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -291,6 +291,13 @@ If there is an issue with attributes being matched correctly you could end up se
...
 if the `clientCA` value set in the [OpenShift master configuration changes](#openshift-master-configuration-changes) step is not the CA that signed the [Create SAML Proxy Client Certificates](#create-saml-proxy-client-certificates) then you could see an infinite redirect between OpenShift oAuth and SAML Proxy or other certificate errors in the browser or various logs.
 
+### User Attributes Missing or Incorrect
+Symptoms:
+Logging into Openshift with the SAML provider will have a 403 Forbidden Error.
+On the saml-auth pod (only the debug version) running on Openshift, you will see this line in `/var/log/httpd/mellon_diagnostics`
+`am_check_permissions processing condition 0 of 1: varname="user" flags=[REG] str=".+" directive="MellonCond user .+ [REG]" failed (no OR condition) returning HTTP_FORBIDDEN`
+This error is an indication that your SAML mappings in Keycloak are not mapping correctly. Ensure that the property `username` from keycloak is mapped to the SAML Attribute Name `user`. This involves setting a mapper in the client in Keycloak. See [testing_with_keycloak.md](keycloak/testing_with_keycloak.md#mapping-the-data-from-keycloak-to-mod_auth_mellon)
```
Comment (itewk) [inline README.md]: @bontreger change all of the "keycloak" references to IdP since this error is independent of the specific IdP

**Review by bontreger — COMMENTED** [review]
Changed it to be more generic, with keycloak as the example

File: README.md [hunk 2]
```diff
@@ -296,7 +296,7 @@ Symptoms:
 Logging into Openshift with the SAML provider will have a 403 Forbidden Error.
 On the saml-auth pod (only the debug version) running on Openshift, you will see this line in `/var/log/httpd/mellon_diagnostics`
 `am_check_permissions processing condition 0 of 1: varname="user" flags=[REG] str=".+" directive="MellonCond user .+ [REG]" failed (no OR condition) returning HTTP_FORBIDDEN`
-This error is an indication that your SAML mappings in Keycloak are not mapping correctly. Ensure that the property `username` from keycloak is mapped to the SAML Attribute Name `user`. This involves setting a mapper in the client in Keycloak. See [testing_with_keycloak.md](keycloak/testing_with_keycloak.md#mapping-the-data-from-keycloak-to-mod_auth_mellon)
+This error is an indication that your SAML mappings coming from your IdP are not mapping correctly. The SAML property needs to be mapped as "user", whereas some identity providers, like keycloak, send this property as "username". See [testing_with_keycloak.md](keycloak/testing_with_keycloak.md#mapping-the-data-from-keycloak-to-mod_auth_mellon)
```
Comment (itewk) [inline README.md]: @bontreger keycloack actually doens't send any properties by default it just expects you will look at the NAME_ID field and use that by default which this code currently doesn't, it expects the `user` property. I would just drop everything from the "whereas some" on.

**Comment by itewk** [comment]
@brenton another for your review

**Pull request merged**